src/components/SigningModals/MultiScreenModal.js

```diff
@@ -0,0 +1,180 @@
+              {({ height }) => (
+                <AnimatedDiv
+                  style={{
+                    position: 'relative',
+                    height: measuredHeight ? height : 'auto',
+                  }}
+                >
+                  <Transition
+                    config={(_, state) =>
+                      state === 'leave' ? springs.instant : springs.smooth
+                    }
+                    items={step}
+                    immediate={firstStart}
+                    from={{
+                      opacity: 0,
+                      transform: `translate3d(${5 * GU * direction}px, 0, 0)`,
+                    }}
+                    enter={{
+                      opacity: 1,
+                      transform: 'translate3d(0, 0, 0)',
+                    }}
+                    leave={{
+                      opacity: 0,
+                      transform: `translate3d(${5 * GU * -direction}px, 0, 0)`,
+                    }}
+                    onRest={(_, status) => {
+                      if (status === 'update') {
+                        setMeasuredHeight(false)
+                      }
+                    }}
+                    onStart={onStart}
+                    native
+                  >
+                    {currentStep => ({ opacity, transform }) => {
+                      // For better performance we avoid reflows between screen changes by matching the screen width with the modal width
+                      const screenWidth =
+                        screens[currentStep].width || DEFAULT_MODAL_WIDTH
+
+                      return (
+                        <AnimatedDiv
+                          ref={elt => {
+                            if (elt) {
+                              setHeight(elt.clientHeight)
+                            }
+                          }}
+                          style={{
+                            position:
+                              currentStep === step ? 'static' : 'absolute',
+                            transform: transform,
+                            opacity: opacity,
+                            width: containWidth
+                              ? viewportWidth - gutter
+                              : screenWidth,
+                            padding: smallMode ? 3 * GU : 5 * GU,
+                          }}
+                        >
+                          {renderScreen(currentStep)}
+                        </AnimatedDiv>
+                      )
+                    }}
+                  </Transition>
+                </AnimatedDiv>
+              )}
+            </Spring>
+          </Modal>
+        )
+      }}
+    </Viewport>
+  )
+}
+
+MultiScreenModal.propTypes = {
+  visible: PropTypes.bool,
+  screens: PropTypes.arrayOf(
+    PropTypes.shape({
+      title: PropTypes.string,
+      content: PropTypes.func,
+      disableClose: PropTypes.bool,
+      width: PropTypes.number,
+    })
+  ),
+  onClose: PropTypes.func,
```

Comment:
Since we don't protect usage of this function inside, let's provide a default prop :)